lib/policyEvaluator/evaluator.js

```diff
@@ -133,11 +133,17 @@ function meetConditions(requestContext, statementCondition, log) {
             // reference_policies_multi-value-conditions.html)
             const keyBasedOnRequestContext =
                 findConditionKey(key, requestContext);
-            // Handle IfExists
+            // Handle IfExists and negation operators
+            const operatorsWithNegation = ['StringNotEquals',
```

Comment:
More functional are coming in Integration.
I did not find one, so I run locally my functional test against Amazon S3 to see how these different operators behaved.
I moved both array outside the two loops.